arm-advisor/2016-07-12-preview/swagger/advisor.json

```diff
@@ -0,0 +1,641 @@
+				}],
+				"responses": {
+					"202": {
+						"description": "Recommendation generation is in progress."
+					},
+					"204": {
+						"description": "Recommendation generation has completed."
+					}
+				},
+				"deprecated": false,
+				"x-ms-examples": {
+					"PollGenerateRecommendationsStatus": {
+						"$ref": "../examples/PollGenerateRecommendationsStatus.json"
+					}
+				}
+			}
+		},
+		"/subscriptions/{subscriptionId}/providers/Microsoft.Advisor/recommendations": {
+			"get": {
+				"tags": ["GetRecommendations"],
+				"description": "Obtains cached recommendations for a subscription. The recommendations are generated/computed by invoking generateRecommendations.",
+				"operationId": "Recommendations_List",
+				"parameters": [{
+					"$ref": "#/parameters/subscriptionIdParameter"
+				},
+				{
+					"$ref": "#/parameters/apiVersionParameter"
+				},
+				{
+					"name": "$filter",
+					"in": "query",
+					"description": "The filter to apply on the recommendations.",
+					"required": false,
+					"type": "string"
+				},
+				{
+					"name": "$top",
+					"in": "query",
+					"description": "The number of recommendations per page if using paged version of this API.",
+					"required": false,
+					"type": "integer"
+				},
+				{
+					"name": "$skipToken",
+					"in": "query",
+					"description": "The page continuation token if using paged version of this API.",
+					"required": false,
+					"type": "string"
+				}],
+				"responses": {
+					"200": {
+						"description": "OK",
+						"schema": {
+							"$ref": "#/definitions/CollectionResponse[ResourceRecommendationBase]"
+						}
+					}
+				},
+				"deprecated": false,
+				"x-ms-pageable": {
+					"nextLinkName": "nextLink"
+				},
+				"x-ms-examples": {
+					"ListRecommendations": {
+						"$ref": "../examples/ListRecommendations.json"
+					}
+				}
+			}
+		},
+		"/providers/Microsoft.Advisor/operations": {
+			"get": {
+				"tags": ["Operations"],
+				"description": "Lists all of the available Advisor REST API operations",
+				"operationId": "Operations_List",
+				"parameters": [{
+					"$ref": "#/parameters/apiVersionParameter"
+				}],
+				"responses": {
+					"200": {
+						"description": "OK",
+						"schema": {
+							"$ref": "#/definitions/CollectionResponse[OperationEntity]"
+						}
+					}
+				},
+				"deprecated": false,
+				"x-ms-pageable": {
+					"nextLinkName": "nextLink"
+				}
+			}
+		},
+		"/{resourceUri}/providers/Microsoft.Advisor/recommendations/{recommendationId}/suppressions/{name}": {
+			"get": {
+				"tags": ["Suppressions"],
+				"description": "Obtains detail of a suppression",
+				"operationId": "Suppressions_Get",
+				"parameters": [{
+					"name": "resourceUri",
+					"in": "path",
+					"description": "The fully qualified arm identifier of the resource for which the recommendation applies.",
+					"required": true,
+					"type": "string"
+				},
+				{
+					"name": "recommendationId",
```

Comment:
Is `recommedationId` unique inside `recommendations`? If so why do you need the information about `resourceUri` of that recommendation?